src/test/java/github/grace000/erversay/MockConnection.java

```diff
@@ -0,0 +1,40 @@
+package github.grace000.erversay;
+
+public class MockConnection implements ConnectionWrapper {
+    private boolean listenForConnectionCalled = false;
+    private boolean acceptConnectionCalled = false;
+    private boolean getIOStreamsCalled = false;
+    private boolean writeResponseCalled = false;
+
+    public void listenForConnection(int port) {
+        listenForConnectionCalled = true;
+    }
+
+    public void serveClient(String response) {
+        acceptConnectionCalled = true;
+        getIOStreamsCalled = true;
+        writeResponseCalled = true;
+    }
+
+    public void close() {
+
+    }
+
+    public boolean isWriteResponseCalled() {
+        return writeResponseCalled;
+    }
+
+    public boolean isGetIOStreamsCalled() {
+        return getIOStreamsCalled;
+    }
+
+    public boolean isAcceptConnectionCalled() {
+        return acceptConnectionCalled;
+    }
+
+    public boolean isListenForConnectionCalled() {
+        return listenForConnectionCalled;
+    }
```

Comment:
Small suggestion: I would omit the `is` portion of these mock function names from the beginning. Just `listenForConnectionCalled` or `listenForConnectionWasCalled`. It follows a common pattern for other mocking libraries.